src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -646,6 +707,106 @@ class HintSection {
   }
 }
 
+class MultipleSelectionHintSection extends HintSection {
+  constructor(sectionOptions, cm) {
+    super(sectionOptions, cm);
+
+    this.container.classList.add('CodeMirror-hints-multiple-selection');
+    this.initCompletionButton(sectionOptions.buttonText);
+  }
+
+  initCompletionButton(buttonText) {
+    this.buttonContainer = createContainer('CodeMirror-hint-section-button-container');
+    this.completionButton = document.createElement('button');
+    this.completionButton.className = 'CodeMirror-hint-section-button';
+    this.completionButton.innerHTML = buttonText;
+    this.completionButton.onclick = this.handleButtonClick;
+
+    this.setCompletionButtonVisability(false);
+    this.buttonContainer.appendChild(this.completionButton);
+    this.container.appendChild(this.buttonContainer);
+  }
+
+  handleButtonClick = () => {
+    CodeMirror.signal(this.cm, 'multiplePick', this.getCheckedItemsIds());
+  }
+
+  createListItemContent(displayText, itemOptions) {
+    if (itemOptions.id && itemOptions.id.includes('any')) {
+      return super.createListItemContent(displayText, itemOptions);
+    }
+
+    const checkBoxContainer = createContainer('CodeMirror-checkbox-container');
+    const labelElement = document.createElement('label');
+    labelElement.innerHTML = displayText;
+    const checkBoxInput = document.createElement('input');
+    checkBoxInput.id = itemOptions.id;
+    checkBoxInput.type = 'checkbox';
+    checkBoxContainer.appendChild(checkBoxInput);
+    checkBoxContainer.appendChild(labelElement);
+
+    checkBoxInput.onchange = () => this.setCompletionButtonEnabling(this.hasSelectedItems());
+
+    return checkBoxContainer;
+  }
+
+  checkItem() {
+    const checkBox = this.geItemCheckBox(this.selectedHintIndex);
+
+    checkBox.checked = !checkBox.checked;
+
+    this.setCompletionButtonEnabling(this.hasSelectedItems());
+  }
+
+  setCompletionButtonVisability(isVisible) {
+    const { classList } = this.buttonContainer;
+
+    if (isVisible) {
+      classList.remove('hidden');
+    } else {
+      classList.add('hidden');
+    }
+  }
+
+  setCompletionButtonEnabling(isEnabled) {
+    const { classList } = this.completionButton;
+
+    if (isEnabled) {
```

Comment:
As far as I know, the class name should be put to the toggle method to add/remove a class name.
 
I'm not sure that it will work in our case. What about the next case:
- The button is disabled because there are no selected items.
- The Backspace is pressed and the interaction is moved to the previous section
- During the moving to the previous sections, the button should be disabled and hidden. So in the proposed solution, the call of the toggle method will enable the button (we will have to add some additional checks in other places)

Do I understand it right?